lib/isolator.rb

```diff
@@ -1,6 +1,43 @@
 # frozen_string_literal: true
 
-require "isolator/version"
+require 'anyway'
+require 'uniform_notifier'
 
-module Isolator # :nodoc:
+require 'isolator/version'
+require 'isolator/notifier'
+require 'isolator/errors'
+require 'isolator/config'
+
+require 'isolator/orm_adapters/active_record_adapter'
+
+require 'isolator/adapters/ethon_adapter'
```

Comment:
Why not using [`sniffer`](https://github.com/aderyabin/sniffer)? It does pretty much the same monkey-patching; we'd better patch it (or use [middlewares](https://github.com/aderyabin/sniffer/issues/28) when they become available).

Reply:
I agree. I don't really understand how can we use `sniffer` here, though. It does entirely different thing, even though approach is very much the same.

Reply:
Maybe, we should start with contributing to sniffer)

The idea is the following (like Sidekiq middleware):

```ruby
Sniffer.middlewares << Module.new
  # called before HTTP request
  def self.call(_req_data)
     raise IsolatorError if Isolator.enabled?
     yield
  end
end
```

/cc @aderyabin

We can rely on other Sniffer features such as whitelist/blacklist. And we should not care about adapters specifics at all.
Otherwise we will eventually re-invent the wheel.

Reply:
@palkan I refactored adapters to remove boilerplate as much as possible. Just as an alternative. I feel like result is quite good, even without `sniffer`. I mean we can still use sniffer for web requests and this approach for background jobs and custom extendedness. Anyway, would love to hear your opinion on this.